pkg/kapp/preflight/registry_test.go

```diff
@@ -106,3 +117,174 @@ func TestRegistryRun(t *testing.T) {
+						if cfg == nil {
+							return errors.New("config should be present")
+						}
+						v, ok := cfg["foo"]
+						if !ok {
+							return errors.New("foo config not present")
+						}
+						if v != "bar" {
+							return errors.New("foo should equal 'bar'")
+						}
+						return nil
+					}, true),
+				},
+			},
+			config: map[string]any{
+				"foo": "bar",
+			},
+		},
+		{
+			name: "preflight checks registered, unexpected config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg == nil {
+							return errors.New("config should be present")
+						}
+						_, ok := cfg["foobar"]
+						if ok {
+							return errors.New("foo config should not present")
+						}
+						return nil
+					}, true),
+				},
+			},
+			config: map[string]any{
+				"foo": "bar",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			for _, check := range tc.registry.known {
+				_ = check.SetConfig(tc.config)
+			}
+			err := tc.registry.Run(nil, nil)
+			if tc.shouldErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestRegistryConfig(t *testing.T) {
+	testCases := []struct {
+		name      string
+		registry  *Registry
+		configErr bool
+		shouldErr bool
+	}{
+		{
+			name: "preflight checks registered, config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg == nil {
+							return errors.New("config should be present")
+						}
+						v, ok := cfg["foo"]
+						if !ok {
+							return errors.New("foo config not present")
+						}
+						if v != "bar" {
+							return errors.New("foo should equal 'bar'")
+						}
+						return nil
+					}, true),
+				},
+			},
+		},
+		{
+			name: "preflight checks registered, unexpected config set, error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg == nil {
+							return errors.New("config should be present")
+						}
+						_, ok := cfg["foobar"]
+						if ok {
+							return errors.New("foobar config should not present")
+						}
+						return nil
+					}, true),
+				},
+			},
+		},
+		{
+			name: "preflight checks registered, unexpected preflight check set, error",
+			registry: &Registry{
+				known: map[string]Check{
+					"otherCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg != nil {
+							return errors.New("config should not be present")
+						}
+						return nil
+					}, true),
+				},
+			},
+			configErr: true,
+		},
+	}
+	configYAML := `---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+preflightRules:
+- name: someCheck
+  config:
+    foo: bar
+`
```

Comment:
IMO we don't need unit tests that include parsing the configuration file and then passing that to the `SetConfig` method. This should either be done where unit tests are done for the config file and only if it makes sense to fit it into the existing unit tests.

Reply:
I actually disagree. This is specifically testing the `registry.SetConfig()` API, which doesn't get invoked by configuration, as we call this API before the `registry.Run()`. This API takes in a full configuration file, and extracts the `preflightRules` bits.

Reply:
I'm not saying we shouldn't test the `registry.SetConfig()` API - actually the opposite, I think we absolutely _should_ be testing that API. I just don't think we need to include the logic of parsing the config file format and instead directly provide the `config.Conf` object that is declaratively defined for each test case.

Reply:
For example, I'm thinking something along these lines:
```go
testCases := []struct {
		name      string
		registry  *Registry
		shouldErr bool
		cfg config.Conf
	}{...}
for _, tc := range testCases {
	t.Run(tc.name, func(t *testing.T) {
		err := tc.registry.SetConfig(tc.cfg)
		require.Equalf(t, tc.shouldErr, err != nil, "Invalid error %v", err)
	})
}
```